pkg/pubsub/adapter/adapter.go

```diff
@@ -127,16 +146,20 @@ func (a *Adapter) convert(ctx context.Context, m transport.Message, err error) (
 		event.SetDataContentType(*cloudevents.StringOfApplicationJSON())
 		event.SetType(v1alpha1.PubSubEventType)
 		event.SetID(uuid.New().String())
+		for k, v := range msg.Attributes {
+			event.SetExtension("attributes-"+k, v)
```

Comment:
docs pending.

Reply:
I thought "-" was not permitted in CloudEvents attributes?

I'm also wondering whether this is reversed when we encode -- it seems like it might be simpler to avoid the prefix altogether.

Reply:
Oh, I see -- you're attempting to pack all the PubSub attributes into a single context attribute called "attributes", but you're doing it by direct construction of HTTP headers rather than by actually storing a map in the extension.

This is probably worth a comment.